src/compiler/scala/tools/nsc/transform/Erasure.scala

```diff
@@ -582,17 +584,16 @@ abstract class Erasure extends InfoTransform
       bridge setInfo (otpe cloneInfo bridge)
       bridgeTarget(bridge) = member
 
-      def sigContainsValueClass = (member.tpe exists (_.typeSymbol.isDerivedValueClass))
-
-      val shouldAdd = (
-            !sigContainsValueClass
-        ||  (checkBridgeOverrides(member, other, bridge) match {
-              case Nil => true
-              case es if member.owner.isAnonymousClass => resolveAnonymousBridgeClash(member, bridge); true
-              case es => for ((pos, msg) <- es) reporter.error(pos, msg); false
-            })
-      )
-
+      val shouldAdd = {
+        val sigContainsValueClass = member.tpe.exists(_.typeSymbol.isDerivedValueClass)
+        def bridgeMayClash = bridge.paramss.exists(_.exists(p => p.tpe =:= ObjectTpe || isTrivialTopType(p.tpe)))
```

Comment:
It seems `Object` is not sufficient?

```scala
➜ sandbox spr 10100
Welcome to Scala 2.13.9-20220730-031046-9de07a7 (OpenJDK 64-Bit Server VM, Java 17.0.1).
Type in expressions for evaluation. Or try :help.

scala> class A; class B extends A

scala> class C1 { def compareTo(a: A): Int = 0 }

scala> trait C2[T <: A]{ def compareTo(t: T): Int }

scala> class C3 extends C1 with C2[B] { def compareTo(b: B): Int = 1 }

scala> (new C3).compareTo(new A)
java.lang.ClassCastException: class A cannot be cast to class B (A and B are in unnamed module of loader scala.tools.nsc.interpreter.IMain$TranslatingClassLoader @76396509)
  at C3.compareTo(<console>:1)
  ... 32 elided
```

Reply:
Now it looks at the params of `other` for a `f(t: T)`.